x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/openai/OpenAiChatCompletionResponseHandler.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.openai;
+
+import org.elasticsearch.xpack.inference.external.http.HttpResult;
+import org.elasticsearch.xpack.inference.external.http.retry.ResponseParser;
+import org.elasticsearch.xpack.inference.external.http.retry.RetryException;
+import org.elasticsearch.xpack.inference.external.request.Request;
+
+public class OpenAiChatCompletionResponseHandler extends OpenAiResponseHandler {
+    public OpenAiChatCompletionResponseHandler(String requestType, ResponseParser parseFunction) {
+        super(requestType, parseFunction);
+    }
+
+    @Override
+    RetryException buildExceptionHandling429(Request request, HttpResult result) {
+        // We don't retry, if the chat completion input is too large
+        return new RetryException(false, buildError(CONTENT_TOO_LARGE, request, result));
```

Comment:
Since I suppose it's possible that the 429 could happen if we hit too many requests per minute, maybe we just return a generic 429 message that we have: `RATE_LIMIT` is the variable name.